classes/services/queryBuilders/PKPStatsListQueryBuilder.inc.php

```diff
@@ -0,0 +1,350 @@
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function timeSegment($timeSegment) {
+		$this->timeSegment = $timeSegment == 'day' ? STATISTICS_DIMENSION_DAY : STATISTICS_DIMENSION_MONTH;
+		return $this;
+	}
+
+	/**
+	 * Set the columns (aggregation level)
+	 * Assumes lookingFor and timeSegment are set
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function columns() {
+		switch ($this->lookingFor) {
+			case 'orderedSubmissions':
+				$this->columns = array(STATISTICS_DIMENSION_SUBMISSION_ID);
+				break;
+			case 'totalSubmissionsStats':
+				$this->columns = array($this->timeSegment, STATISTICS_DIMENSION_ASSOC_TYPE);
+				break;
+			case 'submissionStats':
+				$this->columns = array($this->timeSegment, STATISTICS_DIMENSION_ASSOC_TYPE, STATISTICS_DIMENSION_FILE_TYPE);
+				break;
+		}
+		return $this;
+	}
+
+	/**
+	 * Set assoc type filter
+	 * Assumes lookingFor ais set
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterByAssocTypes() {
+		switch ($this->lookingFor) {
+			case 'orderedSubmissions':
+			case 'totalSubmissionsStats':
+			case 'submissionStats':
+				$this->filters[STATISTICS_DIMENSION_ASSOC_TYPE] = array(ASSOC_TYPE_SUBMISSION, ASSOC_TYPE_SUBMISSION_FILE);
+				break;
+		}
+		return $this;
+	}
+
+	/**
+	 * Set submission IDs filter
+	 *
+	 * @param $submissionIds array | int
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterBySubmissionIds($submissionIds) {
+		$this->filters[STATISTICS_DIMENSION_SUBMISSION_ID] = $submissionIds;
+		return $this;
+	}
+
+	/**
+	 * Set section IDs filter
+	 *
+	 * @param $sectionIds array | int
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterBySectionIds($sectionIds) {
+		$this->filters[STATISTICS_DIMENSION_PKP_SECTION_ID] = $sectionIds;
+		return $this;
+	}
+
+	/**
+	 * Set daily date range (from and to) filter
+	 * Assumes that either start or end date exists
+	 *
+	 * @param $dateStart string optional
+	 * @param $dateEnd string optional
+	 *
+	 * @return \PKP\Services\QueryBuilders\PKPStatsListQueryBuilder
+	 */
+	public function filterByDateRange($dateStart = null, $dateEnd = null) {
+		// pre-assumption: one of the dates exist
+		assert($dateStart != null || $dateEnd != null);
+
+		$from = $to = null;
+		// convert the date strings to the right format
+		if ($dateStart && preg_match('/(\d{4})-(\d{2})-(\d{2})/', $dateStart, $matches) === 1) {
+			$from = $matches[1] . $matches[2] . $matches[3];
+		}
+		if ($dateEnd && preg_match('/(\d{4})-(\d{2})-(\d{2})/', $dateEnd, $matches) === 1) {
+			$to = $matches[1] . $matches[2] . $matches[3];
+		}
+		// set the default value for from and to date, if missing
+		if ($to == null) {
+			$to = date('Ymd', time());
+		} elseif ($from == null) {
+			// TO-DO eventually: chose the start date differently ?
+			$from = '20010101';
```

Comment:
You mean to introduce that constant in general?

Reply:
Yeah, even if it's only used once, I think it's good to avoid throwing a number in there directly. And we may reuse it more down the line (issue stats, for example).